### PR TITLE
Selenium multithread cross-browser task refactored

### PR DESCRIPTION
**Pull request by rstoliarsolvd**

### CODE REVIEW

File: src/main/java/com/amazon/pages/AbstractPage.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class AbstractPage {
+
+    private static final Logger LOGGER = Logger.getLogger(AbstractPage.class);
+
+    //    WebDriver driver;
```
Comment (mverkhautsova) [inline src/main/java/com/amazon/pages/AbstractPage.java]: Please if you comment something in your code, add additional comment, why it needed and not removed. Or remove useless code

File: src/main/java/com/amazon/pages/FilterMenuPage.java [hunk 2]
```diff
@@ -23,7 +23,7 @@ public class FilterMenuPage extends AbstractPage{
     @FindBy(xpath = "//div[@class ='nav-sprite hmenu-close-icon']")
     private WebElement closeFilterMenuBtn;
 
-//    @FindBy(xpath = "//*[@id='hmenu-content']")
+    //    @FindBy(xpath = "//*[@id='hmenu-content']")
```
Comment (mverkhautsova) [inline src/main/java/com/amazon/pages/FilterMenuPage.java]: the same as previous comment

File: src/main/java/com/amazon/pages/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    private WebElement desktopCardLayout;
+
+    public HomePage(RemoteWebDriver driver) {
+        super(driver);
+        PageFactory.initElements(driver, this);
+    }
+
+    public boolean isHomePageOpen() {
+//        new WebDriverWait(driver, Duration.ofSeconds(6)).until(ExpectedConditions.visibilityOf(desktopBannerStripe));
+        FilterMenuPage filterMenuPage = new FilterMenuPage(driver);
```
Comment (mverkhautsova) [inline src/main/java/com/amazon/pages/HomePage.java]: the same as previous comment

File: src/main/java/com/amazon/pages/TodaysDealPage.java [hunk 4]
```diff
@@ -28,32 +29,39 @@ public class TodaysDealPage extends AbstractPage{
...
+        LOGGER.info("Verify if goods with deals discounts present on the page: " + areGoodDealsPresent);
+        return areGoodDealsPresent;
     }
 
     public boolean ifTDPageIsOpen() {
 
-//        new WebDriverWait(driver, Duration.ofSeconds(5)).until(ExpectedConditions.visibilityOf(discountGoods.get(1))); //no Fail variant
-        new WebDriverWait(driver, Duration.ofSeconds(5)).until(ExpectedConditions.visibilityOf(header));//for fail screenshot
+        new WebDriverWait(driver, Duration.ofSeconds(5)).until(ExpectedConditions.visibilityOf(discountGoods.get(1))); //no Fail variant
+//        new WebDriverWait(driver, Duration.ofSeconds(5)).until(ExpectedConditions.visibilityOf(header));//for fail screenshot
```
Comment (mverkhautsova) [inline src/main/java/com/amazon/pages/TodaysDealPage.java]: the same as previous comment

File: src/test/testng_suites/parallel-testng.xml [hunk 5]
```diff
@@ -1,18 +1,17 @@
 <!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
 
+<!--<suite name="Parallel-test-method Suite" parallel="methods" thread-count="1">-->
+<suite name="Parallel-test-method Suite" parallel="tests" thread-count="3">
+    <!--    <test verbose="2" preserve-order="true" name="/Users/rstoliar/IdeaProjects/SeptemberTasks/src/main/java/amazon">-->
```
Comment (mverkhautsova) [inline src/test/testng_suites/parallel-testng.xml]: the same as previous comment

File: src/main/java/com/amazon/services/GoTo.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+     *
+     * @param driver
+     * @return
+     */
+    public static HomePage goHome(RemoteWebDriver driver) {
+        UpTab upTab = new UpTab(driver);
+        upTab.clickHomeBtn();
+        return new HomePage(driver);
+    }
+}
```
Comment (mverkhautsova) [inline src/main/java/com/amazon/services/GoTo.java]: It's will be better to name this service f.e. NavigationService.
Also you can not extend Abstract page. And remove constructor of class.

**Review by mverkhautsova — COMMENTED** [review]
Good work! But I left a few comments, plz review them and fix. If you disagree with something, let me know.

File: src/main/java/com/amazon/pages/AbstractPage.java (hunk 1 above)
Comment (rstoliarsolvd) [reply src/main/java/com/amazon/pages/AbstractPage.java]: Description added

File: src/main/java/com/amazon/pages/FilterMenuPage.java (hunk 2 above)
Comment (rstoliarsolvd) [reply src/main/java/com/amazon/pages/FilterMenuPage.java]: Description added

File: src/main/java/com/amazon/pages/HomePage.java (hunk 3 above)
Comment (rstoliarsolvd) [reply src/main/java/com/amazon/pages/HomePage.java]: this was removed

File: src/main/java/com/amazon/pages/TodaysDealPage.java (hunk 4 above)
Comment (rstoliarsolvd) [reply src/main/java/com/amazon/pages/TodaysDealPage.java]: Here comments are already present

Comment (rstoliarsolvd) [reply src/main/java/com/amazon/pages/TodaysDealPage.java]: But it improved now

File: src/test/testng_suites/parallel-testng.xml (hunk 5 above)
Comment (rstoliarsolvd) [reply src/test/testng_suites/parallel-testng.xml]: description added

File: src/main/java/com/amazon/services/GoTo.java (hunk 6 above)
Comment (rstoliarsolvd) [reply src/main/java/com/amazon/services/GoTo.java]: Done

**Comment by marinagavrilenko** [comment]
Thank you for updatings by comments. You can merge this PR.

**Pull request merged**